### PR TITLE
Add braces plugin

### PR DESCRIPTION
**Pull request by klipstein**
Finally finished the braces plugin. 

You can configure `braces` with the following settings:
- `braces: "{\n"`: changes all `{` into `if(foo){`
- `braces: " {\n"`: changes all `{` into `if(foo) {` (of course you could add more spaces, but who wants that :))
- `braces: "\n{"`: changes all `{` into `if(foo)\n{`

### CODE REVIEW

File: lib/plugins/braces.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      }
+      // closing braces
+      if(typeof(namedType) !== 'undefined') {
+        token.twin.value = '\n' + braceDetail.indent + '}';
+      }
+    }
+  });
+  parser = Parser.parse(Stringifier.stringify(parser));
+  wtree = parser.tokenizer.wtree;
+  //console.log(JSON.stringify(getTreeString(wtree)));
```
Comment (seidtgeist) [inline lib/plugins/braces.js]: Can you remove that pl0x?

File: lib/plugins/braces.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+};
+
+function getStripPositions(wtree, braceToken) {
+  var leftTokenPos;
+  var rightTokenPos;
+  var index = leftTokenPos = rightTokenPos = braceToken.tokposw;
+  while(wtree[leftTokenPos--] && wtree[leftTokenPos] &&
+      (wtree[leftTokenPos].name === 9 /*WHITE_SPACE*/ || wtree[leftTokenPos].name === 10 /*LINETERMINATOR*/)) {};
+  if(! wtree[leftTokenPos]) {
```
Comment (seidtgeist) [inline lib/plugins/braces.js:56]: 1999 called, they want to donate their spaces to this line :) Or is the diff view wrongly indenting?

File: lib/plugins/braces.js (hunk 1 above)
Comment (klipstein) [inline lib/plugins/braces.js]: You mean the reparsing part?

File: lib/plugins/braces.js (hunk 2 above)
Comment (klipstein) [inline lib/plugins/braces.js:56]: Sure? Look again. Still sure? The while is empty.

Looks right to me :)

**Pull request merged**